src/Plugin/Field/FieldWidget/ProductVariationSkuWidget.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Drupal\commerce\Plugin\Field\FieldWidget;
+
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Field\Plugin\Field\FieldWidget\StringTextfieldWidget;
+
+/**
+ * Plugin implementation of the 'commerce_auto_sku' widget.
+ *
+ * @FieldWidget(
+ *   id = "commerce_auto_sku",
+ *   label = @Translation("Commerce auto SKU"),
+ *   field_types = {
+ *     "string"
+ *   }
+ * )
+ */
+class ProductVariationSkuWidget extends StringTextfieldWidget {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultSettings() {
+    return array(
```

Comment:
What I think the array shortcut syntax need to be used carefully in the code. Obviously, it is more appropriate in the places like this:

   public function myFunc (array $my_array = []) {}

But it looks like more readable in a multiline array as in the snippet above. Just because the word "array" is clearly highlighted whereas brackets not.

The same says [Drupal coding standard](https://www.drupal.org/docs/develop/standards/coding-standards#array).  [Discussion](https://www.drupal.org/node/2135291) will never stop on this issue.